### PR TITLE
Moved rule group checking to rules evaluation function. Fixed some imports and missing is types.

### PR DESCRIPTION
**Pull request by AntonCroner**

### CODE REVIEW

File: Rule/Rule/Base.ts [hunk 1]
```diff
@@ -13,14 +13,20 @@ export interface Base {
 export namespace Base {
 	export const kinds = ["authorization", "outbound", "inbound"] as const
```
Comment (SiFalorSum) [inline Rule/Rule/Base.ts:14]: I think we should declare constants in their namespaces, exported as `<namespace>.values`

File: Rule/Rule/Base.ts [hunk 2]
```diff
@@ -13,14 +13,20 @@ export interface Base {
...
+	export namespace Kind {
+		export const type = isly.string<Kind>(kinds)
+	}
 	export const categories = ["fincrime", "product", "customer"] as const
 	export type Category = typeof categories[number]
+	export namespace Category {
+		export const type = isly.string<Category>(categories)
+	}
 	export const type = isly.object<Base>({
 		code: isly.string(new RegExp(/^[a-z0-9\-_]+$/)),
```
Comment (SiFalorSum) [inline Rule/Rule/Base.ts:25]: `new RegExp` is unnecessary, since the argument is already a regex pattern, so we can probably clean this up.

**Review by SiFalorSum — CHANGES_REQUESTED** [review]
Some minor cleanup

File: Rule/Rule/Base.ts (hunk 2 above)
Comment (AntonCroner) [reply Rule/Rule/Base.ts]: \+

File: Rule/Rule/Base.ts (hunk 1 above)
Comment (AntonCroner) [reply Rule/Rule/Base.ts]: \+

**Review by SiFalorSum — APPROVED** [review]

**Pull request merged**